### DISCUSSION TITLE
How to use other encoding of env file that is  different from os encoding?

### DISCUSSION THREAD

**Discussion by yzqdev** [encode/starlette#2016] [Q&A]
My OS is windows, and its encoding is GBK. when I use startlette's config module to read a file `.env` whose encoding is utf-8,it says,
```
UnicodeDecodeError: 'gbk' codec can't decode byte 0xab in position 614: illegal multibyte sequence
```
.
and i found the following code,`with open(file_name) as input_file`,without a encoding.
https://github.com/encode/starlette/blob/9450edf28208ba139c2162b38e6ea40eb04bc90b/starlette/config.py#L116-L119

so what can i do to solve the problem that use my gbk windows to read a utf-8 env file?

**Comment by seaskydl** [comment]
This should be a bug

**Reply by Kludex** [reply]
Seems like a bug yep. PR welcome.

**Reply by secrett2633** [reply]
Fix PR : https://github.com/Kludex/starlette/pull/2996

**Comment by injae-kim** [comment]
Fix PR: https://github.com/encode/starlette/pull/2992 👍
